src/plugin/plugin.ts

```diff
@@ -0,0 +1,320 @@
+   * @param {EntitlementsSuccessCallback} callback Callback triggered after a successful getEntitlements call. It will receive an structure of entitlements.
+   * @param {ErrorCallback} errorcallback Callback triggered after an error or when retrieving entitlements.
+   */
+  public static getEntitlements(callback: any, errorcallback: any) {
+    window.cordova.exec(
+      callback,
+      errorcallback,
+      PLUGIN_NAME,
+      "getEntitlements",
+      []
+    );
+  }
+
+  /**
+   * Success callback used when retrieving products.
+   *
+   * @callback ProductsSuccessCallback
+   * @param {[Product]} products - Array containing the product objects.
+   */
+
+  /**
+   * Fetch the product info
+   * @param {[String]} productIdentifiers Array of product identifiers
+   * @param {ProductsSuccessCallback} callback Callback triggered after a successful getProducts call. It will receive an array of product objects.
+   * @param {ErrorCallback} errorcallback Callback triggered after an error or when retrieving products
+   * @param {String} type Optional type of products to fetch, can be inapp or subs. Subs by default
+   */
+  public static getProducts(
+    productIdentifiers: string[],
+    callback: any,
+    errorcallback: any,
+    type: string = "subs"
+  ) {
+    window.cordova.exec(
+      callback,
+      errorcallback,
+      PLUGIN_NAME,
+      "getProductInfo",
+      [productIdentifiers, type]
+    );
+  }
+
+  /**
+   * Success callback used when making a purchase. This callback will be triggered after a successful purchase.
+   * Cancelled purchases will trigger an error callback.
+   *
+   * @callback makePurchaseSuccessCallback
+   * @param {string} productIdentifier - The product identifier of the purchased product.
+   * @param {object} purchaserInfo - The updated Purchaser Info.
+   */
+
+  /**
+   * Error callback triggered when making a purchase. Cancelled purchases will also trigger this callback and
+   * error.userCancelled will be true
+   *
+   * @callback makePurchaseErrorCallback
+   * @param {object} error - The error containing message, code, domain and if user cancelled the purchase.
+   * @param {Boolean} userCancelled - Wether the user cancelled the purchase or not.
+   */
+
+  /**
+   * Make a purchase
+   * @param {string} productIdentifier The product identifier of the product you want to purchase.
+   * @param {makePurchaseSuccessCallback} callback Callback triggered after a successful purchase.
+   * @param {makePurchaseErrorCallback} errorcallback Callback triggered after an error or when the user cancels the purchase.
+   * If user cancelled, userCancelled will be true
+   * @param {String?} oldSku Optional sku you wish to upgrade from.
+   * @param {String} type Optional type of product, can be inapp or subs. Subs by default
+   */
+  public static makePurchase(
+    productIdentifier: string,
+    callback: any,
+    errorcallback: any,
+    oldSku: string | null,
+    type: string = "subs"
+  ) {
```

Comment:
is there a way to do overloads in typescript? so that we can join this and the previous function into one

Reply:
not really